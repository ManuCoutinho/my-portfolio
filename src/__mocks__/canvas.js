HTMLCanvasElement.prototype.getContext = () => {
  return {
    fillRect: jest.fn()
  }
}
