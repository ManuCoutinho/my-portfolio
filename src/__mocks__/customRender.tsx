import { ReactElement, FC, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
