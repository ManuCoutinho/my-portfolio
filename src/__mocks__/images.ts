export const mockImg = [
	{
		imgPng: '/img1.png',
		imgWebp: '/img1.webp',
		alt: 'img1',
		id: 'img1'
	},
	{
		imgPng: '/img2.png',
		imgWebp: '/img2.webp',
		alt: 'img2',
		id: 'img2'
	}
]
