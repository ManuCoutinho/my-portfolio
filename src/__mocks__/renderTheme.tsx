import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
