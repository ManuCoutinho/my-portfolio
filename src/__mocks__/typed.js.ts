/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-empty-function */

export default class Typed {
  constructor(elementId: string, options: any) {}
  toggle(): void {}
  stop(): void {}
  start(): void {}
  destroy(): void {}
  reset(restart?: boolean): void {}
}
