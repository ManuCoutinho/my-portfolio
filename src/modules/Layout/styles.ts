import styled, { css } from 'styled-components'
import { fadeInAnimation } from 'styles/mixins/animations'

export const Main = styled.main`
	${() => css`
		${fadeInAnimation};
	`}
`
