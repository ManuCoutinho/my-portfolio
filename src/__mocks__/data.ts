import { mockImg } from './images'
export default {
  description: 'a beutiful test',
  site: 'http://www.somesit.com',
  repo: 'http://www.somerepo.com',
  styles: 'styles',
  api: 'mock',
  framework: 'next',
  name: 'project',
  tools: ['tools'],
  id: 'id',
  doc: 'http://www.storybook.com',
  coverDefaultLight: 'img.png',
  coverDefaultDark: '',
  coverAltLight: '',
  coverAltDark: '',
  alt: 'mock img',
  img: mockImg,
  route: '/test'
}
