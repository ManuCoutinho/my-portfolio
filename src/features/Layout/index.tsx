import dynamic from 'next/dynamic'
import { Main } from './styles'

const Header = dynamic(() => import('features/Header'), { ssr: false })
const Footer = dynamic(() => import('components/Footer'), { ssr: false })

function Layout({ children }: { children: React.ReactNode }): JSX.Element {
	return (
		<Main>
			<Header />
			{children}
			<Footer />
		</Main>
	)
}

export default Layout
