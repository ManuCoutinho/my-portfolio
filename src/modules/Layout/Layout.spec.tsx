import { screen, waitFor } from '@testing-library/react'
import { render } from '__mocks__/customRender'
import mockRouter from 'next-router-mock'
import Layout from '.'

jest.mock('next/router', () => require('next-router-mock'))
describe('<Layout />', () => {
	beforeEach(() => {
		mockRouter.setCurrentUrl('/')
		mockRouter.asPath
	})
	it('should render the Layout component correctly and match to snapshot', async () => {
		const { container } = render(
			<Layout>
				<p>children</p>
			</Layout>
		)
		await waitFor(() =>
			expect(screen.getByRole('navigation')).toBeInTheDocument()
		)
		expect(screen.getByText('children')).toBeInTheDocument()

		expect(screen.getByRole('contentinfo')).toBeInTheDocument()
		expect(container).toMatchSnapshot()
	})
})
