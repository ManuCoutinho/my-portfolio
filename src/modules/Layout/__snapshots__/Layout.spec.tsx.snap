// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render the Layout component correctly and match to snapshot 1`] = `
<div>
  <main
    class="sc-dde9e527-0 qAIZd"
  >
    <header
      class="sc-4d9dc00-0 kRTWky"
    >
      <nav
        aria-label="Main Menu"
        class="sc-4d9dc00-1 bZitzR"
        role="navigation"
      >
        <h1
          class="sc-4d9dc00-3 cpeVBn"
        >
          <a
            aria-label="logo Manu Coutinho"
            href="#"
          >
            Manu
            <span>
              Coutinho_
            </span>
          </a>
        </h1>
        <ul
          class="sc-4d9dc00-2 gynxJE"
        >
          <li
            aria-label="toggle theme"
            role="menuitem"
          >
            <button
              aria-label="switch theme"
              class="sc-58177537-0 DDfXa"
              role="switch"
              type="button"
            >
              <iconify-icon
                icon="ph:sun-dim"
              />
            </button>
          </li>
          <li
            aria-label="toggle language"
            role="menuitem"
          >
            <button
              class="sc-58177537-0 sc-8025e36d-0 DDfXa ivBpje"
            >
              <iconify-icon
                icon="ph:globe-light"
              />
            </button>
          </li>
          <li
            aria-label="toggle drawer"
            role="menuitem"
          >
            <button
              aria-controls="menu"
              aria-expanded="false"
              aria-haspopup="true"
              aria-hidden="true"
              aria-label="Abrir Menu"
              class="sc-72623f9b-1 hmKOkO"
              role="button"
            >
              <iconify-icon
                icon="ph:list-bold"
              />
            </button>
          </li>
        </ul>
        <aside
          class="sc-72623f9b-0 lkwMCQ"
          data-state="close"
        >
          <ul
            aria-label="options"
            class="sc-3bcf75e5-0 llAXfA"
            id="menu"
            role="menu"
          >
            <li
              aria-owns="menu"
              class="sc-3bcf75e5-1 bWMXSD"
              data-state="off"
              role="menuitem"
            >
              <a
                class="sc-3bcf75e5-2 jcQJNi"
                href="/home#home"
              >
                Home
              </a>
            </li>
            <li
              aria-owns="menu"
              class="sc-3bcf75e5-1 bWMXSD"
              data-state="off"
              role="menuitem"
            >
              <a
                class="sc-3bcf75e5-2 jcQJNi"
                href="/home#about"
              >
                Sobre
              </a>
            </li>
            <li
              aria-owns="menu"
              class="sc-3bcf75e5-1 bWMXSD"
              data-state="off"
              role="menuitem"
            >
              <a
                class="sc-3bcf75e5-2 jcQJNi"
                href="/home#portfolio"
              >
                Projetos
              </a>
            </li>
            <li
              aria-owns="menu"
              class="sc-3bcf75e5-1 bWMXSD"
              data-state="off"
              role="menuitem"
            >
              <a
                class="sc-3bcf75e5-2 jcQJNi"
                href="/home#articles"
              >
                Artigos
              </a>
            </li>
            <li
              aria-owns="menu"
              class="sc-3bcf75e5-1 bWMXSD"
              data-state="off"
              role="menuitem"
            >
              <a
                class="sc-3bcf75e5-2 jcQJNi"
                href="/home#contact"
              >
                Contato
              </a>
            </li>
          </ul>
        </aside>
      </nav>
    </header>
    <p>
      children
    </p>
    <footer
      class="sc-31e02b67-0 hUwWXt"
      data-testid="footer"
    >
      <p>
        Copyright Â© 2023 Todos os direitos reservados. 
      </p>
      <p>
        Designed by 
        <span>
          Manu Coutinho.
        </span>
      </p>
    </footer>
  </main>
</div>
`;
